boards/arduino-leonardo/src/pins.rs

```diff
@@ -10,12 +10,23 @@ avr_hal_generic::impl_board_pins! {
         portc: crate::atmega32u4::PORTC,
         portd: crate::atmega32u4::PORTD,
         porte: crate::atmega32u4::PORTE,
-        // Not used yet
-        // portf: crate::atmega32u4::PORTF,
+        portf: crate::atmega32u4::PORTF,
     }
 
     /// Reexport of the Leonardo's pins, with the names they have on the board
     pub struct Pins {
+        /// `A0`
+        pub a0: portf::pf7::PF7,
+        /// `A1`
+        pub a1: portf::pf6::PF6,
+        /// `A2`
+        pub a2: portf::pf5::PF5,
+        /// `A3`
+        pub a3: portf::pf4::PF4,
+        /// `A4`
+        pub a4: portf::pf1::PF1,
+        /// `A5`
+        pub a5: portf::pf0::PF0,
```

Comment:
Please also document the alternate functions for these pins.  I'd also move this block to the end, so the digital pins are first in the documentation.

```suggestion
        /// `A0`
        ///
        /// * `ADC7` channel
        pub a0: portf::pf7::PF7,
        /// `A1`
        ///
        /// * `ADC6` channel
        pub a1: portf::pf6::PF6,
        /// `A2`
        ///
        /// * `ADC5` channel
        pub a2: portf::pf5::PF5,
        /// `A3`
        ///
        /// * `ADC4` channel
        pub a3: portf::pf4::PF4,
        /// `A4`
        ///
        /// * `ADC1` channel
        pub a4: portf::pf1::PF1,
        /// `A5`
        ///
        /// * `ADC0` channel
        pub a5: portf::pf0::PF0,
```

Reply:
done & done (ADCx are the alternate functions?)

Reply:
Correct, by default the pins are Digital I/O, and they can be reconfigured as ADC channels.